src/model/physics/cushion.ts

```diff
@@ -71,6 +71,8 @@ export class Cushion {
     if (futurePosition.y < -TableGeometry.tableY) {
       return Cushion.bounceIn(Math.PI / 2, ball)
     }
+
+    return null;
```

Comment:
Should we throw an error here actually? Is that "unhandled condition" even possible?

Reply:
Yes throwing some unreachable state error would be best.

The way it is called prevents it from reaching the error state (table.ts)
```
   if (Cushion.willBounce(a, t)) {
      const incidentSpeed = Cushion.bounce(a, t)
```

Reply:
I went on to refactor that code to use undefined. Let me know if its worse :)